### PR TITLE
Test: add some tests for utils

### PR DESCRIPTION
**Pull request by SamSalvatico**
Added unit tests and moved folders into a `__test__` structure

Moved `formatResponse` and `formatError` to a separated utils file to make it easier to test them

### CODE REVIEW

File: src/__tests__/client/utils/client-utils.test.ts [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+  data?: never;
+  error: T;
+  response: Response;
+};
+
+interface ApiError {
+  message: string;
+  code: string;
+}
+
```
Comment (matteolc) [inline src/__tests__/client/utils/client-utils.test.ts:147]: Maybe move this types in a separate file (just for types?)

**Review by matteolc — APPROVED** [review]
LGTM just minor comment

**Review by alewin — APPROVED** [review]
LGTM

File: src/__tests__/client/utils/client-utils.test.ts (hunk 1 above)
Comment (SamSalvatico) [reply src/__tests__/client/utils/client-utils.test.ts]: done, thank you @matteolc

**Pull request merged**